### PR TITLE
[datadog_integration_azure] Enable Resource collection by default for new customers

### PR DESCRIPTION
**Pull request by ashwingandhi-ddog**
https://datadoghq.atlassian.net/browse/AZINTS-2316

Update example on terraform [page](https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/integration_azure) with `resource_collection_enabled` set to true.

PRs for other approaches to create an azure integration:
- https://github.com/DataDog/datadog-serverless-functions/pull/767
- https://github.com/DataDog/web-ui/pull/130720

### CODE REVIEW

File: docs/resources/integration_azure.md [hunk 1]
```diff
@@ -15,15 +15,16 @@ Provides a Datadog - Microsoft Azure integration resource. This can be used to c
...
+  tenant_name                 = "<azure_tenant_name>"
+  client_id                   = "<azure_client_id>"
+  client_secret               = "<azure_client_secret_key>"
+  host_filters                = "examplefilter:true,example:true"
+  app_service_plan_filters    = "examplefilter:true,example:another"
+  container_app_filters       = "examplefilter:true,example:one_more"
+  automute                    = true
+  cspm_enabled                = true
+  custom_metrics_enabled      = false
+  resource_collection_enabled = true
```
Comment (ava-silver) [inline docs/resources/integration_azure.md:27]: nit: can we put this above `cspm_enabled` in the example? just for organization

**Review by ava-silver — CHANGES_REQUESTED** [review]
i think we also need to update this: https://github.com/DataDog/terraform-provider-datadog/blob/efc5c793a795651f13581f44ada694b54269ce43/datadog/fwprovider/resource_datadog_azure_integration.go#L96

**Review by estherk15 — APPROVED** [review]

**Comment by ashwingandhi-ddog** [comment]
Created an integration without passing `resource_collection_enabled` and the app registration was created with Resource Collection enabled.

https://ricky-test.datadoghq.com/integrations/azure?search=azure&clientId=b7721f8d-d466-4fcb-8140-f5e0259f4edd&panel=resources&tab=configuration&tenantId=4d3bac44-0230-4732-9e70-cc00736f0a97

**Review by ava-silver — APPROVED** [review]

**Review by estherk15 — APPROVED** [review]

**Comment by ashwingandhi-ddog** [comment]
Closing out this PR. 
 "The Computed + Optional fields basically mean if not set by the user, the backend will set the value".
 We are defaulting this to `true` on the backend.

**Comment by ashwingandhi-ddog** [comment]
Closing out this PR.
"The Computed + Optional fields basically mean if not set by the user, the backend will set the value".
We are defaulting this to true on the backend.

**Pull request closed without merging**